### PR TITLE
[MRG+1] MAINT moved all "d" modules to pytest.

### PR DESCRIPTION
**Pull request by NelleV**
This PR migrates the modules starting by "d" from nosetests to pytest.

### CODE REVIEW

**Comment by QuLogic** [comment]
`test_dviread` is covered by #7872.

**Comment by NelleV** [comment]
I removed the corresponding commit.

File: lib/matplotlib/tests/test_dates.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import six
```
Comment (efiring) [inline lib/matplotlib/tests/test_dates.py]: The line below uses six.

Comment (QuLogic) [inline lib/matplotlib/tests/test_dates.py]: As an import?

Comment (NelleV) [inline lib/matplotlib/tests/test_dates.py]: I don't see where the import is useful.

Comment (dopplershift) [inline lib/matplotlib/tests/test_dates.py]: Yeah, I'm not able to find any use of `six.` in that file.

Comment (efiring) [inline lib/matplotlib/tests/test_dates.py]: Sorry, I wasn't thinking properly.

**Comment by phobson** [comment]
Used Github's UI to resolve a merge conflict that came from merging the test_c* PR before this one. I'll let the CI run and then finish the merge.

**Pull request merged**